drivers/display/display.c

```diff
@@ -0,0 +1,217 @@
+	dev->controller_ops = param->controller_ops;
+
+	ret = dev->controller_ops->init(dev);
+
+	*device = dev;
+
+	return ret;
+}
+
+/****************************************************************************//**
+ * @brief Frees the resources allocated by display_init().
+ *
+ * @param dev - The device structure.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_remove(struct display_dev *device)
+{
+	int32_t ret;
+
+	ret = spi_remove(device->spi_desc);
+
+	ret |= gpio_remove(device->dc_pin);
+	ret |= gpio_remove(device->reset_pin);
+
+	free(device);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Turns display on/off.
+ *
+ * @param device - The device structure.
+ *        on_off - DISP_ON
+ *                 DISP_OFF
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_on_off(struct display_dev *device, uint8_t on_off)
+{
+
+	return device->controller_ops->display_on_off(device, on_off);
+}
+
+/***************************************************************************//**
+ * @brief Moves cursor to desired position.
+ *
+ * @param device - The device structure.
+ *        row    - row
+ *        column - column
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_move_cursor(struct display_dev *device, uint8_t row,
+			    uint8_t column)
+{
+
+	return device->controller_ops->move_cursor(device, row, column);
+}
+
+/***************************************************************************//**
+ * @brief Prints character at selected position.
+ *
+ * @param device - The device structure.
+ *        row    - row
+ *        column - column
+ *        ascii  - corresponding number to ascii table
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_print_ascii(struct display_dev *device, uint8_t ascii,
+			    uint8_t row, uint8_t column)
+{
+
+	return device->controller_ops->display_print_by_ascii(device, ascii, row,
+			column);
+}
+
+/***************************************************************************//**
+ * @brief Clears data on display.
+ *
+ * @param device - The device structure.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_clear(struct display_dev *device)
+{
+	int32_t ret = SUCCESS;
+	uint8_t i, j;
+
+	for(i = 0; i < device->rows_nb; i++)
+		for(j = 0; j < device->cols_nb; j++)
+			ret |= display_print_ascii(device, ' ', i, j);
```

Comment:
even if you've probably seen ret |= ...; in no-OS code, this is not ok - that's old code
 quit if there is an error